content/guides/end-to-end-testing/auth0-authentication.md

```diff
@@ -111,10 +117,146 @@ require('dotenv').config()
 }
 ```
 
+Note that `auth0_client_secret` is only needed for
+[programmatic login](#Programmatic-Login).
+
 :::
 
 ## Custom Command for Auth0 Authentication
 
+There are two ways you can authenticate to Auth0:
+
+- [Login with `cy.origin()`](#Login-with-cy-origin)
+- [Programmatic Login](#Programmatic-Login)
+
+### Login with [`cy.origin()`](/api/commands/origin)
+
+Next, we'll write a custom command called `loginToAuth0` to perform a login to
+[Auth0](https://auth0.com). This command will use
+[`cy.origin()`](/api/commands/origin) to
+
+1. Navigate to the Auth0 login
+2. Input user credentials
+3. Sign in and redirect back to the
+   [Cypress Real World App](https://github.com/cypress-io/cypress-realworld-app)
+4. Cache the results with [`cy.session()`](/api/commands/session)
+
+```js
+// cypress/support/auth-provider-commands/auth0.ts
+
+Cypress.Commands.add('loginToAuth0', (username: string, password: string) => {
+  const log = Cypress.log({
+    displayName: 'AUTH0 LOGIN',
+    message: [`🔐 Authenticating | ${username}`],
+    // @ts-ignore
+    autoEnd: false,
+  })
+  log.snapshot('before')
+
+  // App landing page redirects to Auth0.
+  cy.visit('/')
+
+  // Login on Auth0 (we'll add caching later).
+  cy.origin(
```

Comment:
I like the idea of making this a `loginToAuth0` function similar to the okta/cognito examples per @emilyrohrbough's suggestion, so the `cy.session` example becomes easier to digest as to what the changes were between this code snippet and consuming `cy.session`

Reply:
I'm good with that if we can come up with a better naming convention. The difference between `loginBy` and `loginTo` is really not clear to me.

Reply:
Not too worried about the `loginBy` vs the `loginTo`, more so how digestible the examples are. What I mean is encapsulating everything in a function like in the [okta example](https://github.com/cypress-io/cypress-documentation/blob/v12/content/guides/end-to-end-testing/okta-authentication.md?plain=1#L101) and using it in the custom command [below](https://github.com/cypress-io/cypress-documentation/blob/v12/content/guides/end-to-end-testing/okta-authentication.md?plain=1#L124). That way when we get to the [session code](https://github.com/cypress-io/cypress-documentation/blob/v12/content/guides/end-to-end-testing/okta-authentication.md?plain=1#L166) we just need to reference the function so its easier for the user to diff